haproxy/README.md

```diff
@@ -69,3 +69,106 @@ This package does not currently support high availability on its own.
 ## Monitoring
 
 This service can be monitored by making TCP connections to the configured port (default is 80)
+
+# Testing haproxy
+
+Since the haproxy does not run in isolation but must be bound to a backend webserver, the following instructions exercise it in a simple scenario, one haproxy and one backend web server.
+
+### Build the haproxy core plan
+
+At a terminal build the core/haproxy
+
+```bash
+hab pkg build .
+
+# copy the last_build.env because it gets overwritten
+# in later steps and is required for testing later
+cp results/last_build.env results/last_build_haproxy.env
+```
+
+### Build and install the test plans
+
+The following sequence of steps will do the following:
+
+* install the haproxy just built above
+* build and install 2 services that depend on and exercise the new haproxy
+* manually verify the working of the 2 services
+
+Enter habitat studio and do the following
+
+```bash
+# install the haproxy just built
+source results/last_build.env
+hab pkg install results/$pkg_artifact
+
+# build the test plans
+build test/haproxy/
+build test/webapp/
+
+# install chef/inspec and dependency core/busybox
+hab pkg install -b core/busybox
+hab pkg install chef/inspec
+
+# load the backend web application first
+source results/last_build_webapp.env
+hab svc load $pkg_origin/$pkg_name
+
+# load the haproxy load balancer
+source results/last_build_testhaproxy.env
+hab svc load $pkg_origin/$pkg_name --bind=backend:webapp.default
+```
+
+Optionally perform the following manual test steps.  Both curl's should return content successfully
+
+```bash
+hab pkg install -b core/curl
+
+# check that the backend webserver is running on port 9090
+curl localhost:9090
+
+# check that the haproxy on port 80 is reverse proxying to the backend webserver
+curl localhost
+```
+
+Remain in the same habitat studio for the next section
+
+### Run the tests
+
+```bash
+hab pkg exec chef/inspec inspec exec ./test
```

Comment:
This causes deprecation warnings and will probably stop working when InSpec 4.0 releases.

```
[28][default:/src:0]# hab pkg exec chef/inspec inspec exec ./test
[DEPRECATION] `processes.list` is deprecated. Please use `processes.entries` instead. It will be removed in version 4.0.
[DEPRECATION] `processes.list` is deprecated. Please use `processes.entries` instead. It will be removed in version 4.0.
```